src/libraries/System.Runtime.Serialization.Formatters/src/System.Runtime.Serialization.Formatters.csproj

```diff
@@ -5,6 +5,10 @@
     <TargetFrameworks>$(NetCoreAppCurrent)</TargetFrameworks>
     <Nullable>enable</Nullable>
   </PropertyGroup>
+  <PropertyGroup>
+    <!-- On certain runtimes BinaryFormatter isn't supported at all. -->
+    <DefineConstants Condition="'$(TargetsBrowser)' == 'true'">$(DefineConstants);BINARYFORMATTER_PNSE</DefineConstants>
```

Comment:
this will need a `$(NetCoreAppCurrent)-Browser` slice in the `<TargetFrameworks>` to work.

Reply:
an alternative would be to mark the whole assembly PNSE if we think that none of the types make sense without working BinaryFormatter.

Reply:
@akoeplinger Marking the entire assembly as PNSE was proposed earlier but rejected. The feedback was that I should use runtime detection if possible. Is there another way to detect at runtime if I'm running in browser without taking a dependency on the new _RuntimeInformation_ assembly? It doesn't even need to follow best practice _per se_. This assembly ships with the runtime so it's allowed to take a dependency on runtime implementation details. It can also access the publics of System.Private.CoreLib.

Reply:
`RuntimeInformation` lives in System.Private.CoreLib so you don't need to take an extra dependency and can do `if (RuntimeInformation.IsOSPlatform(OSPlatform.Browser))`.

We've disabled whole assemblies with PNSE on Browser in a lot of cases, I'm not totally sure why this one should be different. Is the previous discussion you mentioned available somewhere?

Reply:
No, it was a phone call, I think with @terrajobst? But the takeaway was that for 5.0 we only approved obsoleting `Serialize` and `Deserialize`. The overall assembly contains considerably more functionality than simply `BinaryFormatter`. There was no intent to obsolete these other types in 5.0 or to prohibit applications from using them. And if a wasm app wants to use them, more power to them, I suppose?